lib/galaxy/datatypes/sequence.py

```diff
@@ -603,16 +603,25 @@ def sniff( self, filename ):
 
         >>> from galaxy.datatypes.sniff import get_test_fname
         >>> fname = get_test_fname( '1.fastqsanger' )
-        >>> Fastq().sniff( fname )
+        >>> FastqSanger().sniff( fname )
         True
         >>> fname = get_test_fname( '2.fastqsanger' )
-        >>> Fastq().sniff( fname )
+        >>> FastqSanger().sniff( fname )
         True
         """
         compressed = is_gzip(filename) or is_bz2(filename)
         if compressed and not isinstance(self, Binary):
             return False
-        headers = get_headers( filename, None )
+        headers = get_headers( filename, None, count=1000 )
+
+        # If this is a FastqSanger-derived class, then check to see if the base qualities match
+        if isinstance(self, FastqSanger) or isinstance(self, FastqSangerGz) or isinstance(self, FastqSangerBz2):
+            if not self.sangerQualities(headers):
+                return False
+        else:
```

Comment:
It took a while, but I think I found why the `else` was needed for binary files, it's because the sniffing order for them is set by the order of registering them. I'm changing that and will retest that it works.